packages/peregrine/lib/store/actions/cart/asyncActions.js

```diff
@@ -31,42 +31,19 @@ export const createCart = () =>
         }
 
         try {
-            const guestCartEndpoint = '/rest/V1/guest-carts';
-            const signedInCartEndpoint = '/rest/V1/carts/mine';
-            const cartEndpoint = user.isSignedIn
-                ? signedInCartEndpoint
-                : guestCartEndpoint;
-
-            const cartId = await request(cartEndpoint, {
-                method: 'POST'
-            });
+            const { data } = await fetchCartId();
```

Comment:
How do we feel about directly using the `useMutation` API within a thunk? Should we implement a `useAwaitMutation`? It'd protect us from changes to the API from apollo hooks later.

@jimbo @tjwiebell @supernova-at

Reply:
The only reason I would propose doing this would be to match the error implementation of `useAwaitQuery`. It looks to me like `useMutation` still resolves on error state, and expects you to destructure `error` to handle that.

```
const { data, error } = await fetchCartId();
const receivePayload = error ? new Error(error) : data.cartId;
dispatch(actions.getCart.receive(receivePayload));
```